src/core/qgsvectorlayer.h

```diff
@@ -2229,6 +2248,9 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
 
     bool accept( QgsStyleEntityVisitorInterface *visitor ) const override;
 
+    //! cache the results of the feature counter
+    void onSymbolsCounted() SIP_SKIP;
```

Comment:
@nyalldawson @m-kuhn when I talked about forcing caching in my comments it was related to exposing this function as public and calling it in the tests. Otherwise travis tests failed since that slot/function was never called despite the connection. I am aware that this is bad practice and should not remain but I wanted to prove that my changes worked properly with the tests that I made. As stated if this is removed, the PR should still work but the travis tests will not be able to properly indicate when something fails as this slot never got triggered.

What do you propose on this issue?

Reply:
@m-kuhn @nyalldawson simple ping to get feedback on this aspect of the code.

Reply:
@m-kuhn @nyalldawson once again a new ping, I know that this change will raise a red flag and I want to know if the ability to test or hiding this slot is more important.